### PR TITLE
Use apiKey if configured for certain requests

### LINKED ISSUE #442: OTP-RR should forward x-api-key to OTP endpoints
From @landonreed:
Some OTP-middleware implementations require that OTP proxied endpoints be protected via the `x-api-key` header that we pass with our plan trip API requests. 
We need otp-react-redux to pass the headers from the `getOtpFetchOptions` method to the fetch that happens within createQueryAction: https://github.com/opentripplanner/otp-react-redux/blob/dev/lib/actions/api.js#L755

### PR DESCRIPTION
**Pull request by evansiroky**
Fixes #442.

### CODE REVIEW

File: lib/actions/api.js [hunk 1]
```diff
@@ -775,6 +765,54 @@ function createQueryAction (endpoint, responseAction, errorAction, options = {})
...
+  const apiKey = config?.persistence?.otp_middleware?.apiKey
+  if (apiKey) {
+    const headers = fetchOptions.headers || {}
+    fetchOptions.headers = {
+      'x-api-key': apiKey,
+      ...headers
+    }
+    if (!fetchOptions.mode) {
+      fetchOptions.mode = 'cors'
+    }
```
Comment (landonreed) [inline lib/actions/api.js:811]: There's a bit of shared code here with `getSecureFetchOptions`. Did you consider any reuse? Given that they inject a lot of the same headers, I think that might be worthwhile.

**Review by landonreed — COMMENTED** [review]
Looks good. I pushed 527c6dc9422326f6cabfc70bf018e78dd65cb1d3 to fix a minor omission, but have a question about reusing some code from `getSecureFetchOptions`.

File: lib/actions/api.js (hunk 1 above)
Comment (evansiroky) [reply lib/actions/api.js:811]: I did consider reuse, but only the apiKey and cors thing are needed, so I figured it wasn't worth the effort to add extra configuration arguments to that method.

Comment (landonreed) [reply lib/actions/api.js:811]: See #445 for some proposed changes.

**Review by landonreed — COMMENTED** [review]
@evansiroky, I just made some changes in #445. Let me know what you think (feel free to merge here and re-assign this PR if they look OK to you).

**Comment by evansiroky** [comment]
Hmm, #445 this looks a bit better.

**Review by landonreed — APPROVED** [review]

**Comment by landonreed** [comment]
@binh-dam-ibigroup, once you approve, could you merge and create a release PR?

**Review by binh-dam-ibigroup — APPROVED** [review]

**Pull request merged**